### PR TITLE
[MM-25657] api4/config: add config migrate endpoint

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
As part of the migrating `mattermost` cli commands to the [mmctl](https://github.com/mattermost/mmctl), this PR is the backend part of migrating `mattermost config migrate` [command](https://docs.mattermost.com/administration/command-line-tools.html?highlight=migrate#mattermost-config-migrate). Added `/config/migrate` endpoint to the `api/v4` so that it can be accessed via REST API. Accessing this functionality requires `manage_system` permissions.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-25657

### CODE REVIEW

**Comment by cpanato** [comment]
can you please describe a bit better the PR? migrate from what to where? not clear to me.

**Comment by isacikgoz** [comment]
@cpanato updated the PR description.

File: api4/config.go [hunk 1]
```diff
@@ -227,3 +228,36 @@ func patchConfig(c *Context, w http.ResponseWriter, r *http.Request) {
...
+	}
+	to, ok := props["to"].(string)
+	if !ok {
+		c.SetInvalidParam("to")
+		return
+	}
+
+	auditRec := c.MakeAuditRecord("migrateConfig", audit.Fail)
+	auditRec.AddMeta("from", from)
+	auditRec.AddMeta("to", from)
```
Comment (wiggin77) [inline api4/config.go:247]: Should be `auditRec.AddMeta("to", to)`

File: model/client4.go [hunk 2]
```diff
@@ -3247,6 +3247,19 @@ func (c *Client4) UpdateConfig(config *Config) (*Config, *Response) {
 	return ConfigFromJson(r.Body), BuildResponse(r)
 }
 
+// MigrateConfig will migrate existing config with the new one.
```
Comment (wiggin77) [inline model/client4.go:3250]: Minor nit:  I suggest `MigrateConfig will migrate existing config to the new one.`

**Comment by justinegeffen** [comment]
@isacikgoz, could you confirm that https://github.com/mattermost/mattermost-server/pull/15506 are the only docs needed for this PR? :) Thank you!

**Comment by cwarnermm** [comment]
Documentation updated via https://github.com/mattermost/docs/pull/4206

**Pull request merged**